torch/csrc/autograd/custom_function.h

```diff
@@ -263,6 +263,14 @@ template <class T>
 template <typename X, typename... Args>
 auto Function<T>::apply(Args&&... args)
     -> std::enable_if_t<std::is_same<X, T>::value, forward_t<X, Args...>> {
+  const auto& functorch_tls = at::functorch::functorchTLSAccessor();
+  if (functorch_tls) {
+    // Function support for functorch is handled in Python.
+    // Here we are dealing with a (C++) Function, which is not supported.
+    // Let's raise an error instead of being silently incorrect.
+    functorch_tls->checkSupportsCppAutogradFunction();
```

Comment:
Nope, we want this for all functorch transforms.

If we vmap over the torch::autograd::Function's forward, it turns out that the backward pass isn't preserved (because the torch::autograd::Function gets invoked on BatchedTensors that cannot have grad_fn).